central/probeupload/manager/manager_impl.go

```diff
@@ -41,151 +40,90 @@ var (
 )
 
 type manager struct {
-	rootDir           string
-	freeDiskThreshold int64
-
-	fsMutex sync.RWMutex
+	rootDir              string
+	blobStore            blobstore.Datastore
+	freeStorageThreshold int64
 }
 
-func newManager(persistenceRoot string) *manager {
+func newManager(datastore blobstore.Datastore) *manager {
 	return &manager{
-		rootDir:           filepath.Join(persistenceRoot, rootDirName),
-		freeDiskThreshold: defaultFreeDiskThreshold,
+		rootDir:              rootBlobPathPrefix,
+		blobStore:            datastore,
+		freeStorageThreshold: defaultFreeStorageThreshold,
 	}
 }
 
-func (m *manager) cleanUpModuleVersionDir(modVer string) error {
-	subDir := filepath.Join(m.rootDir, modVer)
-	subDirEntries, err := os.ReadDir(subDir)
+func (m *manager) getAllProbeBlobs() ([]string, error) {
+	names, err := m.blobStore.GetNames(sac.WithAllAccess(context.Background()))
 	if err != nil {
-		return errors.Wrap(err, "could not read module version subdirectory")
-	}
-
-	hasFiles := false
-
-	for _, subDirEnt := range subDirEntries {
-		if subDirEnt.Name() == "." || subDirEnt.Name() == ".." {
-			continue
-		}
-
-		hasFiles = true
-
-		if !subDirEnt.IsDir() {
-			log.Warnf("Unexpected non-directory entry %q in probe upload directory for module version %s", subDirEnt.Name(), modVer)
-			continue
-		}
-		if strings.HasPrefix(subDirEnt.Name(), tempUploadPrefix) {
-			tempUploadDir := filepath.Join(subDir, subDirEnt.Name())
-			log.Infof("Removing leftover temporary upload directory %q", tempUploadDir)
-			if err := os.RemoveAll(tempUploadDir); err != nil && !os.IsNotExist(err) {
-				log.Warnf("Failed to remove leftover temporary upload directory %q: %v", tempUploadDir, err)
-			}
-		}
-		if !probeupload.IsValidProbeName(subDirEnt.Name()) {
-			log.Warnf("Unexpected non-probe entry %q in probe upload directory for module version %s", subDirEnt.Name(), modVer)
-			continue
-		}
+		return nil, err
 	}
-
-	if !hasFiles {
-		log.Infof("Removing empty module version directory %q", subDir)
-		if err := os.Remove(subDir); err != nil && !os.IsNotExist(err) {
-			log.Warnf("Failed to remove empty module version directory %q", subDir)
+	var blobs []string
+	for _, name := range names {
```

Comment:
Yes, I thought about it. An alternative way is to add search capability so we can search the qualified entries. I will add it in another PR which could change this part. Searching should be more generic and flexible way.

Reply:
Yeah I was thinking we could do something like map[string]string so it would be like: "type": "collector probe" or something unstructured which would just be generally flexible